src/handler/page.rs

```diff
@@ -74,14 +76,23 @@ impl PageInner {
         execute(cmd, self.sender.clone(), Some(self.session_id.clone())).await
     }
 
-    /// This responds with the final http response when the page is loaded
-    pub(crate) async fn wait_for_navigation(&self) -> Result<Option<Arc<HttpRequest>>> {
-        let (tx, rx) = oneshot_channel();
-        self.sender
-            .clone()
-            .send(TargetMessage::WaitForNavigation(tx))
-            .await?;
-        Ok(rx.await?)
+    /// Create a PDL command future
+    pub(crate) fn execute_future<T: Command>(&self, cmd: T) -> Result<CommandFuture<T>> {
```

Comment:
this does to actually execute the future, right?

so perhaps we rename this?

Reply:
It does not execute it, just returns a future for later manual execution using `.await`, I will rename the function to `command_future`, let me know if you have a better name 🙇